File: math/7_ReverseInteger/Solution.go
import (
	"math"
)

func reverse(x int) int {
    xReversed := 0
    
    for x != 0 {
        xReversed = (xReversed * 10) + (x % 10)
        x = x / 10
    }

    if xReversed < math.MinInt32 || xReversed > math.MaxInt32 {
        return 0
    }
    return int(xReversed)
}

File: math/7_ReverseInteger/Solution.java
class Solution {
    public int reverse(int x) {
        long xReversed = 0;
		while(x != 0) {
			xReversed = xReversed * 10 + x % 10; 
            if (xReversed < Integer.MIN_VALUE || xReversed > Integer.MAX_VALUE) {
                return 0;
            }
            x /= 10;
		}
		return (int) xReversed;	
    }
}
